File: app/src/main/java/cm/aptoide/pt/home/apps/model/AppcUpdateApp.java
package cm.aptoide.pt.home.apps.model;

public class AppcUpdateApp extends UpdateApp {

  private boolean hasPromotion;
  private float appcReward;

  public AppcUpdateApp(String name, String md5, String icon, String packageName, int progress,
      boolean isIndeterminate, String version, int versionCode, Status updateStatus, long appId,
      boolean hasPromotion, float appcReward) {
    super(name, md5, icon, packageName, progress, isIndeterminate, version, versionCode,
        updateStatus, appId);
    this.hasPromotion = hasPromotion;
    this.appcReward = appcReward;
  }

  public boolean hasPromotion() {
    return hasPromotion;
  }

  public float getAppcReward() {
    return appcReward;
  }

  @Override public Type getType() {
    return Type.APPC_MIGRATION;
  }
}

File: app/src/main/java/cm/aptoide/pt/home/apps/model/AppcUpdateApp.kt
package cm.aptoide.pt.home.apps.model

import cm.aptoide.pt.home.apps.App

data class AppcUpdateApp(val name: String,
                         override val md5: String,
                         val icon: String,
                         override val packageName: String,
                         override val progress: Int,
                         val version: String,
                         override val versionCode: Int,
                         override var status: StateApp.Status? = null,
                         val appId: Long,
                         val hasPromotion: Boolean,
                         val appcReward: Float) : StateApp {


  override fun getIdentifier(): String {
    return md5
  }

  override fun getType(): App.Type {
    return App.Type.APPC_MIGRATION
  }
}